Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -1741,6 +1743,35 @@ void WebProcessProxy::requestTermination(ProcessTerminationReason reason)
     processDidTerminateOrFailedToLaunch(reason);
 }
 
+void WebProcessProxy::acquireServiceWorkerTerminationAssertion(WebCore::ServiceWorkerIdentifier identifier)
+{
+    WEBPROCESSPROXY_RELEASE_LOG(ProcessSuspension, "acquireServiceWorkerTerminationAssertion: Acquiring a background assertion to terminate service worker %llu", identifier.toUInt64());
+
+    m_serviceWorkersWithTerminationActivities.add(identifier);
+    m_dropServiceWorkerTerminationActivityTimer.startOneShot(WebCore::SWServer::defaultTerminationDelay + 1_s);
```

Comment:
Do we really need this?
If the timer kicks in, it means stopping failed. In which case, the network process proxy should be notified of the unresponsiveness and `disableRemoteWorkers` be called.
How about just dropping the activity within `disableRemoteWorkers`?

Reply:
Will do